File: GUIA/d29_main_application_project/peticiones_ws/peticiones_ws-db/src/main/java/mx/com/telcel/di/sds/gsac/dsc/ws/main_application/peticiones_ws/db/model/PeticionProperties.java
package mx.com.telcel.di.sds.gsac.dsc.ws.main_application.peticiones_ws.db.model;

import java.util.List;
import mx.com.telcel.di.sds.gsac.dsc.ws.main_application.catalogs.db.M2kCatTransaccionesFront;
import mx.com.telcel.di.sds.gsac.dsc.ws.main_application.core.db.Area;
import mx.com.telcel.di.sds.gsac.dsc.ws.main_application.core.db.informix.M2kCatUsuarios;
import mx.com.telcel.di.sds.gsac.dsc.ws.main_application.peticiones_ws.db.entities.AplicativoTelcel;

/**
 *
 * @author Juan
 */
public class PeticionProperties {

    private List<M2kCatUsuarios> listM2kCatUsuarios;
    private List<M2kCatTransaccionesFront> listM2kCatTransaccionesFront;
    private List<Area> listArea;
    private UsuarioProperties usuarioProperties;
    private List<AplicativoTelcel> listAplicativo;

    public List<M2kCatUsuarios> getListM2kCatUsuarios() {
        return listM2kCatUsuarios;
    }

    public void setListM2kCatUsuarios(List<M2kCatUsuarios> listM2kCatUsuarios) {
        this.listM2kCatUsuarios = listM2kCatUsuarios;
    }

    public List<M2kCatTransaccionesFront> getListM2kCatTransaccionesFront() {
        return listM2kCatTransaccionesFront;
    }

    public void setListM2kCatTransaccionesFront(List<M2kCatTransaccionesFront> listM2kCatTransaccionesFront) {
        this.listM2kCatTransaccionesFront = listM2kCatTransaccionesFront;
    }

    public List<Area> getListArea() {
        return listArea;
    }

    public void setListArea(List<Area> listArea) {
        this.listArea = listArea;
    }

    public UsuarioProperties getUsuarioProperties() {
        return usuarioProperties;
    }

    public void setUsuarioProperties(UsuarioProperties usuarioProperties) {
        this.usuarioProperties = usuarioProperties;
    }

    public List<AplicativoTelcel> getListAplicativo() {
        return listAplicativo;
    }

    public void setListAplicativo(List<AplicativoTelcel> listAplicativo) {
        this.listAplicativo = listAplicativo;
    }

    @Override
    public String toString() {
        return "PeticionProperties{" + "listM2kCatUsuarios=" + listM2kCatUsuarios + ", listM2kCatTransaccionesFront=" + listM2kCatTransaccionesFront + ", listArea=" + listArea + ", usuarioProperties=" + usuarioProperties + ", listAplicativoTelcel=" + listAplicativo + '}';
    }
}

File: GUIA/d29-main-application/src/app/peticiones-ws/model/PeticionProperties.ts
import { Response } from "@angular/http";
import { M2kCatTransaccionesFront, toM2kCatTransaccionesFrontList } from "../../catalogos/model/M2kCatTransaccionesFront";
import { Area, toAreaList } from "../../core/model/Area";
import { M2kCatUsuarios, toM2kCatUsuariosList } from "../../core/model/M2kCatUsuarios";
import { AplicativoTelcel, toAplicativoTelcelList } from "./AplicativoTelcel";
import { UsuarioProperties, toUsuarioProperties } from "./UsuarioProperties";

export class PeticionProperties {
  constructor(
    public listM2kCatUsuarios: M2kCatUsuarios[],
    public listM2kCatTransaccionesFront: M2kCatTransaccionesFront[],
    public listArea: Area[],
    public usuarioProperties: UsuarioProperties,
    public listAplicativo: AplicativoTelcel[]
  ) { }

  static getNewInstance(): PeticionProperties {
    return new PeticionProperties([], [], [], null, []);
  }
}

export function toPeticionProperties(r: any): PeticionProperties {
  let output: PeticionProperties = null;
  if (r) {
    output = <PeticionProperties>({
      listM2kCatUsuarios: toM2kCatUsuariosList(r.listM2kCatUsuarios),
      listM2kCatTransaccionesFront: toM2kCatTransaccionesFrontList(r.listM2kCatTransaccionesFront),
      listArea: toAreaList(r.listArea),
      usuarioProperties: toUsuarioProperties(r.usuarioProperties),
      listAplicativo: toAplicativoTelcelList(r.listAplicativo)
    });
  }
  return output;
}

export function mapPeticionProperties(response: Response) {
  let responseJson = response.json();
  let output: PeticionProperties = PeticionProperties.getNewInstance();
  if (responseJson) {
    output = toPeticionProperties(responseJson);
  }
  return output;
}
